services/director-v2/tests/unit/with_swarm/test_modules_dynamic_sidecar_docker_api.py

```diff
@@ -392,6 +402,28 @@ async def test_dynamic_sidecar_in_running_state_and_node_id_is_recovered(
     assert dynamic_sidecar_state == (ServiceState.RUNNING, "")
 
 
+async def test_dynamic_sidecar_get_dynamic_sidecar_sate_fail_to_schedule(
+    dynamic_sidecar_service_spec: Dict[str, Any],
+    dynamic_sidecar_settings: DynamicSidecarSettings,
+    cleanup_test_dynamic_sidecar_service: None,
+    docker_swarm: None,
+) -> None:
+    _inject_impossible_resources(dynamic_sidecar_service_spec)
+    service_id = await docker_api.create_service_and_get_id(
+        dynamic_sidecar_service_spec
+    )
+    assert service_id
+
+    # wait for the service to get scheduled
+    await asyncio.sleep(0.2)
+
+    dynamic_sidecar_state = await docker_api.get_dynamic_sidecar_state(service_id)
+    assert dynamic_sidecar_state == (
+        ServiceState.PENDING,
+        "no suitable node (insufficient resources on 1 node)",
```

Comment:
ok, there will be a lot of comments here.

I think the text should be more informative. Also I am not too sure about users knowing anything about what a node is.
Basically I see 2 use-cases:
1. the one you test here. you have a service with impossible resource needs (can happen if a specific mpi node is down for example)
2. the other where there is some load and the nodes cannot take care of that service right now, but later

So first thing, can you differentiate between both use-cases?

Then I would propose to change the message in something like:
```not enough computational resources available to run NODE_NAME (missing resource X and Y)```

Reply:
Yes, I am aware of the cryptic nature of the messages. We really have this bad habit of exposing docker information directly. Also during failures of the service.

In order to distinguish the two proposed cases I'd have to try and do some sort of string matching on the error field. Not feeling too confident about it and how good it might be with edge cases.
Instead I would add an extra string in front of the docker error, like:

```
There was an issue scheduling your service. Please inform support and provide the following message:  `SERVUCE_ID` reported  `no suitable node (insufficient resources on 1 node)`
```

Reply:
Ok, so first if I understand well we currently have both use-cases in 1 right?
so I guess most of the time we are in the case 2. where the resource will be available at a later time point. Therefore I would modify your message like so:
```The service {service_name} is waiting for available resources. Please inform support and provide the following message in case this does not resolve in a few minutes:  `SERVICE_ID` reported  `no suitable node available` ```